kernel/src/engine/arrow_utils.rs

```diff
@@ -275,7 +279,39 @@ fn get_indices(
     // field, and info about where it appears in the requested_schema, or None if the field is not
     // requested
     let all_field_info = fields.iter().enumerate().map(|(parquet_index, field)| {
-        let field_info = requested_schema.fields.get_full(field.name());
+        let field_id: Option<i64> = field
+            .metadata()
+            .get(PARQUET_FIELD_ID_META_KEY)
+            .and_then(|x| x.parse::<i64>().ok());
+        println!("Field id is: {field_id:?} for schema: {:?}", field.name());
+        // How can we be sure that the field name in the requested schema == the arrow schema we
+        // got form parquet metadata? Parquet metadata may only have physical name. Or it may only
+        // be identifiable by column id?
+        //
+        let field_info = if using_id {
+            requested_schema
+                .fields
+                .iter()
```

Comment:
We should probably iterate the `requested_schema` once and build a map of the ids, otherwise this is `n^2` in the size of the schema

Reply:
Done! PTAL